course/src/isometry.cc

```diff
@@ -160,23 +153,9 @@ namespace ekumen
 
     Isometry operator*(const Isometry &i, const Isometry &i2)
     {
-        Isometry m;
-        Matrix3 p = i.matrix3_isometry;
-        Matrix3 q = i2.matrix3_isometry;
-
-        Vector3 a, b;
-
-        for (size_t i = 0; i < 3; ++i)
-        {
-            a = p.row(i);
-            for (size_t j = 0; j < 3; ++j)
-            {
-                b = q.col(j);
-                m.matrix3_isometry[i][j] = a.dot(b);
-            }
-        }
-        m.vector3_isometry = i.vector3_isometry + i2.vector3_isometry;
-        return m;
+        Matrix3 m1 = m1.mul(i.matrix3_isometry, i2.matrix3_isometry);
```

Comment:
Notice here what I meant about the definition of the `mul()` method. Your using the `m1` instance, which is not yet initialized, to be able to run `mul()` on two other matrixes, and then store the result in `m1`. Please, refactor the method so that this works like
```
Matrix3 m1{i.matrix3_isometry.mul(i2.matrix3_isometry)};
```

Reply:
fixed and refactored.